provider/pkg/gen/schema.go

```diff
@@ -346,12 +346,32 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 			if idPath == "" {
 				idPath = dd.getMethod.Path
 			}
-			v, err := g.buildIdParams(typeName, idPath, inputProperties, &response)
+			var queryParams strings.Builder
+			for param, details := range dd.getMethod.Parameters {
+				if details.Location != "query" || !details.Required {
+					continue
+				}
+				if queryParams.Len() == 0 {
+					_, err = queryParams.WriteString("?")
+					if err != nil {
+						return err
+					}
+				} else {
+					_, err = queryParams.WriteString("&")
+					if err != nil {
+						return err
+					}
+				}
+				queryParams.WriteString(fmt.Sprintf("%[1]s={%[1]s}", param))
```

Comment:
This feels a bit low-level. Consider applying something like https://github.com/pulumi/pulumi-google-native/blob/2da011f7ce69fcf7e2c05f2f2ce0777617b31f2a/provider/pkg/provider/http.go#L227 instead.

Reply:
I restructured it a bit. LMK what you think.